pkg/controllers/provisioning/v1alpha1/reallocator/filter.go

```diff
@@ -0,0 +1,36 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reallocator
+
+import (
+	"context"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Filter struct {
+	kubeClient client.Client
+}
+
+// Gets Nodes fitting some underutilization
+// TODO: add predicate
+func (f *Filter) GetUnderUtilizedNodes(ctx context.Context) ([]*v1.Node, error) {
+	return []*v1.Node{}, nil
+}
+
+// Get Pods scheduled to a node
+func (f *Filter) getPodsOnNode(ctx context.Context) ([]*v1.Pod, error) {
```

Comment:
unclear why this method is private, yet the other is public. If they're meant to be used by the outside world, make them public.